tests/sniffer.py

```diff
@@ -7,36 +7,89 @@
+            return self.captures[test_name]
+        except BaseException as ex:
+            return None
+
+    def _simplify_packets(self, packets):
+        def _simplify(x):
+            x['src_mac'] = x['_source']['layers']['eth']['eth.src']
+            x['dst_mac'] = x['_source']['layers']['eth']['eth.dst']
+            x['time_epoch'] = float(x['_source']['layers']['frame']['frame.time_epoch'])
+            x['time'] = x['_source']['layers']['frame']['frame.time']
+            x['ieee1905_layer'] = x['_source']['layers']['ieee1905']
+            x['msg_type'] = int(x['ieee1905_layer']['ieee1905.message_type'])
+            x['mid'] = int(x['ieee1905_layer']['ieee1905.message_id'], 16)
+            x['tlvs'] = {entry[0]: entry[1] for entry in filter(
+                lambda a: a[0].startswith('ieee1905.') == False, x['ieee1905_layer'].items())}
+            return x
+        return list(map(_simplify, packets))
+
+    def get_packets_for_cmdu_type(self, test_name, msg_type: int):
+        def filter_func(x): return x['msg_type'] == msg_type
+        try:
+            return list(filter(filter_func, self.get_packet_capture(test_name)))
+        except:
+            err("No packets captured for specified cmdu type: {}".format(msg_type))
+            return None
+        return list(filter(filter_func, self.get_packet_capture(test_name)))
 
-    def start(self, outputfile_basename):
-        '''Start tcpdump if enabled by config.'''
-        if opts.tcpdump:
-            debug("Starting tcpdump, output file {}.pcap".format(outputfile_basename))
-            os.makedirs(os.path.join(opts.tcpdump_dir, 'logs'), exist_ok=True)
-            outputfile = os.path.join(opts.tcpdump_dir, outputfile_basename) + ".pcap"
-            command = ["tcpdump", "-i", self.interface, "-w", outputfile]
-            self.tcpdump_proc = subprocess.Popen(command, stderr=subprocess.PIPE)
-            # tcpdump takes a while to start up. Wait for the appropriate output before continuing.
-            # poll() so we exit the loop if tcpdump terminates for any reason.
-            while not self.tcpdump_proc.poll():
-                line = self.tcpdump_proc.stderr.readline()
-                debug(line.decode()[:-1])  # strip off newline
-                if line.startswith(b"tcpdump: listening on " + self.interface.encode()):
-                    # Make sure it doesn't block due to stderr buffering
-                    self.tcpdump_proc.stderr.close()
-                    break
-            else:
-                err("tcpdump terminated")
-                self.tcpdump_proc = None
+    def filter_packets_by_time(self, packets, start_time=None, end_time=time()):
+        def _get_time_epoch_for_packet(packet):
+            t = packet['time_epoch']
+            if start_time:
+                return t >= start_time and t <= end_time
+            return t <= end_time
+        return list(filter(_get_time_epoch_for_packet, packets))
```

Comment:
Same here, I would do this with a list comprehension
```python
def filter_packets_by_time(self, packets, start_time=None, end_time=None):
    if not end_time:
        end_time = time()
    return [x for x in packets if x['time_epoch'] > start_time and x['time_epoch'] < end_time]
```
This also illustrates an important python pitfall: never use function calls for default arguments:
```
>>> def test_time(t=time()):
...     print(t)
...
>>> time()
1587480434.0399244
>>> test_time()
1587480426.4636228
>>> time()
1587480438.1276047
```
See how it appears we've gone back in time? That's because the default argument is evaluated when the python interpreter first sees the function, NOT when it's called!